File: leo2-node/src/main/java/org/deku/leo2/node/messaging/auth/AuthorizationHandler.java
package org.deku.leo2.node.messaging.auth;

import org.deku.leo2.node.auth.IdentityConfiguration;
import org.deku.leo2.node.messaging.auth.v1.AuthorizationMessage;
import sx.jms.Handler;

import javax.jms.JMSException;
import javax.jms.Session;

/**
 * Created by masc on 30.06.15.
 */
public class AuthorizationHandler implements Handler<AuthorizationMessage> {
    @Override
    public void onMessage(AuthorizationMessage message, Session session) throws JMSException {
        IdentityConfiguration.instance().getIdentity().setId(message.getId());
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/messaging/handlers/AuthorizationHandler.kt
package org.deku.leoz.node.messaging.handlers

import org.deku.leoz.node.messaging.entities.AuthorizationMessage
import sx.jms.Channel
import sx.jms.Handler

/**
 * Created by masc on 30.06.15.
 */
class AuthorizationHandler : Handler<AuthorizationMessage> {
    override fun onMessage(message: AuthorizationMessage, replyChannel: Channel?) {
        // TODO: Push authorization update handling. May revoke the node's authorization key
    }
}
